scheduler/src/cook/kubernetes/compute_cluster.clj

```diff
@@ -189,16 +189,14 @@
 
 (defn determine-cook-expected-state-on-startup
   "We need to determine everything we should be tracking when we construct the cook expected state. We should be tracking
-  all tasks that are in the running state as well as all pods in kubernetes. We're given an already existing list of
-  all running tasks entities (via (->> (cook.tools/get-running-task-ents)."
-  [conn api-client compute-cluster-name running-tasks-ents]
+  all tasks that are in the running state as well as all pods in kubernetes. We query for a list of
+  all running tasks entities."
+  [conn api-client compute-cluster-name]
   (let [db (d/db conn)
         [_ pod-name->pod] (api/try-forever-get-all-pods-in-kubernetes api-client compute-cluster-name)
         all-tasks-ids-in-pods (into #{} (keys pod-name->pod))
         _ (log/debug "All tasks in pods (for initializing cook expected state): " all-tasks-ids-in-pods)
-        running-tasks-in-cc-ents (filter
-                                   #(-> % cook.task/task-entity->compute-cluster-name (= compute-cluster-name))
-                                   running-tasks-ents)
+        running-tasks-in-cc-ents (map #(d/entity db %) (cc/get-job-instance-ids-for-cluster-name db compute-cluster-name))
```

Comment:
Other way is probably faster, because you query the DB once and filter N times. This way you query N times and filter N times.

Reply:
this way you query and pass in a filter N times. The filter and query happens in the database

But overall, this happens just at startup.  Later if you add just one cluster it queries for its own instances

Reply:
Is querying for one cluster's instances faster than getting all instances or mostly the same?
Are we happy leaving the initial list of instances in memory for the lifetime of cook?

having each cluster get its own instances seems cleaner, but I can change it if we expect a significant startup overhead

Reply:
I was thinking of waiting jobs, not running jobs. This isn't an issue.